ledger/internal/appcow.go

```diff
@@ -478,24 +478,24 @@ func (cb *roundCowState) StatefulEval(gi int, params *logic.EvalParams, aidx bas
 		return false, transactions.EvalDelta{}, ledgercore.LogicEvalError{Err: err, Details: details}
 	}
 
-	// If program passed, build our eval delta, and commit to state changes
-	if pass {
-		// Before Contract to Contract calls, use BuildEvalDelta because it has
-		// hairy code to maintain compatibility over some buggy old versions
-		// that created EvalDeltas differently.  But after introducing c2c, it's
-		// "too late" to build the EvalDelta here, since the ledger includes
-		// changes from this app and any inner called apps. Instead, we now keep
-		// the EvalDelta built as we go, in app evaluation.  So just use it.
-		if cb.proto.LogicSigVersion < 6 {
-			evalDelta, err = calf.BuildEvalDelta(aidx, &params.TxnGroup[gi].Txn)
-			if err != nil {
-				return false, transactions.EvalDelta{}, err
-			}
-			evalDelta.Logs = params.TxnGroup[gi].EvalDelta.Logs
-			evalDelta.InnerTxns = params.TxnGroup[gi].EvalDelta.InnerTxns
-		} else {
-			evalDelta = params.TxnGroup[gi].EvalDelta
+	// Before Contract to Contract calls, use BuildEvalDelta because it has
+	// hairy code to maintain compatibility over some buggy old versions
+	// that created EvalDeltas differently.  But after introducing c2c, it's
+	// "too late" to build the EvalDelta here, since the ledger includes
+	// changes from this app and any inner called apps. Instead, we now keep
+	// the EvalDelta built as we go, in app evaluation.  So just use it.
+	if cb.proto.LogicSigVersion < 6 {
```

Comment:
I find it scary to change this code for the benefit of dry-run.  The evaldelta is already in `params.TxnGroup[gi].EvalDelta` anyway. Could that be used somehow, instead of changing anything in the ledger package?

Reply:
> Could that be used somehow, instead of changing anything in the ledger package?

This is a good point, I haven't thought about it before but using the actual `ep` makes sense too - I'll try to go down this route as well, thanks!